docs/MPURegionsUsageGuide.md

```diff
@@ -0,0 +1,324 @@
+	3. Access permissions
+		1. **RO** - ReadOnly
+		2. **RW** - Read/Write
+		3. **No Access**
+	4. Execute permissions
+		1. **X** - Executable
+		2. **XN** - Execute Never
+
+- For configuring an MPU region, we need to set below ARM MPU registers:
+	1. **MPU_RNR** : MPU Region Number register
+	2. **MPU_RBAR** : MPU Region Base Address register
+	3. **MPU_RASR** : MPU Region Attribute and Size Register
+
+- MPU_RNR is a unified identifier to the MPU system. This value should not exceed max number of MPU regions for the processor.
+- MPU_RBAR defines the starting address of a MPU region.
+- MPU_RASR encodes size of MPU region and attributes like read/write access, execute permission etc.
+
+- For overlapping memory ranges b/w MPU regions, attributes of highest-numbered MPU region will overwrite those set by lower MPU regions for that overlapping address range.
+
+## Prerequisites for MPU Usage in TizenRT
+
+- In TizenRT, support for configuring MPU regions is available only for Protected Build.
+- Provided MPU is supported in a chip, the following configs need to be enabled for configuring MPU regions(for ARMv7M architecture):
+
+```
+	---------------------------------------------------------------------------------
+	CONFIG FLAG			VALUE		DESCRIPTION
+	---------------------------------------------------------------------------------
+	CONFIG_BUILD_PROTECTED		y		TizenRT Protected build
+	CONFIG_ARMV7M_MPU		y		MPU support
+	CONFIG_ARMV7M_MPU_NREGIONS	16		Number of MPU regions
+	---------------------------------------------------------------------------------
+```
+
+- Similarly, for ARMv8M architecture, corresponding MPU configs should be enabled.
+- In addition to these essential configs, below configs will decide which MPU region is used for what purpose:
+
+```
+	--------------------------------------------------------------------------------------------
+	CONFIG FLAG				TYPE		DESCRIPTION
+	--------------------------------------------------------------------------------------------
+	CONFIG_APP_BINARY_SEPARATION		Boolean		Application binary separation build
+	CONFIG_SUPPORT_COMMON_BINARY		Boolean		Support for Common Binary
+	CONFIG_MPU_STACK_OVERFLOW_PROTECTION	Boolean		MPU Stack Overflow Protection
+	CONFIG_OPTIMIZE_APP_RELOAD_TIME		Boolean		Optimizations for app reload time
+	---------------------------------------------------------------------------------------------
+```
+
+## Functions used and Enumeration for MPU regions in TizenRT
+
+- In case of no App-Binary Separation, the following 2 functions are used for configuring MPU regions for User Internal SRAM and User Flash.
+
+```c
+/****************************************************************************
+ * Name: mpu_user_flash
+ *
+ * Description:
+ *   Configure a region for user program flash
+ *
+ ****************************************************************************/
+static inline void mpu_userflash(uint32_t region, uintptr_t base, size_t size)
+{
+.
+.
+.
+}
```

Comment:
I think we can remove the empty function body and give just the prototypes.

Reply:
Function prototype should be the correct way. Also, mention details about the input parameter

Reply:
done